onedal/linear_model/tests/test_linear_regression.py

```diff
@@ -147,3 +148,96 @@ def test_reconstruct_model(queue, dtype):
 
     tol = 1e-5 if res.dtype == np.float32 else 1e-7
     assert_allclose(gtr, res, rtol=tol)
+
+
+@pytest.mark.parametrize("queue", get_queues("cpu"))
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.parametrize("fit_intercept", [False, True])
+@pytest.mark.skipif(
+    not daal_check_version((2025, "P", 1)),
+    reason="Functionality introduced in later versions",
+)
+def test_overdetermined_system(queue, dtype, fit_intercept):
+    gen = np.random.default_rng(seed=123)
+    X = gen.standard_normal(size=(10, 20))
```

Comment:
`multivariate_normal.rvs` from `scipy` might be more suitable in this case

Reply:
You could say that it'd make a more reasonable target to want to run multiple regression on, but from the test's perspective, that shouldn't matter as long as the LHS matrix (which doesn't use the response variables) is not positive-definite.